### PR TITLE
feat(PORTAL-2536): support EIP1559

### PR DESCRIPTION
**Pull request by zctocm**
### support EIP1559 transaction

new RPCs:

- eth_estimate1559Fee
- eth_maxPriorityFeePerGas

changed RPCs:
- eth_feeHistory

big changes in the `background`:

- eth_signTransaction

big changes in the `front-end`:

- estimate-tx

---
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/helios/1146)

### CODE REVIEW

File: .yarn/versions/7dea276f.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+releases:
+  "@fluent-wallet/estimate-tx": patch
+
+declined:
+  - helios-background
+  - "@fluent-wallet/consts"
+  - "@fluent-wallet/eth-transaction-schema"
```
Comment (yqrashawn) [inline .yarn/versions/7dea276f.yml:7]: this should be patch

File: .yarn/versions/7dea276f.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+releases:
+  "@fluent-wallet/estimate-tx": patch
+
+declined:
+  - helios-background
+  - "@fluent-wallet/consts"
```
Comment (yqrashawn) [inline .yarn/versions/7dea276f.yml:6]: this should be patch

File: .yarn/versions/7dea276f.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  - "@fluent-wallet/eth-transaction-schema"
+  - helios-popup
+  - "@fluent-wallet/eth_estimate-1559-fee"
```
Comment (yqrashawn) [inline .yarn/versions/7dea276f.yml:9]: this should be patch

File: packages/rpcs/eth_maxPriorityFeePerGas/package.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "@fluent-wallet/eth_max-priority-fee-per-gas",
+  "type": "module",
+  "main": "index.js",
+  "version": "0.0.1",
```
Comment (yqrashawn) [inline packages/rpcs/eth_maxPriorityFeePerGas/package.json:5]: better version new pkgs at 0.0.0 and use yarn version to upgrade them

File: .yarn/versions/7dea276f.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  - "@fluent-wallet/eth_fee-history"
+  - "@fluent-wallet/eth_max-priority-fee-per-gas"
+  - "@fluent-wallet/eth_sign-transaction"
+  - "@fluent-wallet/wallet_add-network"
+  - "@fluent-wallet/wallet_add-vault"
+  - "@fluent-wallet/wallet_export-account-group"
+  - "@fluent-wallet/wallet_send-transaction"
+  - "@fluent-wallet/spec"
+  - "@fluent-wallet/tx-history-checker"
+  - "@fluent-wallet/shorten-address"
```
Comment (yqrashawn) [inline .yarn/versions/7dea276f.yml:27]: there're pkgs that not changed in this commit

**Review by yqrashawn — CHANGES_REQUESTED** [review]

File: .yarn/versions/7dea276f.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  "@fluent-wallet/estimate-tx": patch
+
+declined:
+  - helios-background
+  - "@fluent-wallet/consts"
+  - "@fluent-wallet/eth-transaction-schema"
+  - helios-popup
+  - "@fluent-wallet/eth_estimate-1559-fee"
+  - "@fluent-wallet/eth_fee-history"
+  - "@fluent-wallet/eth_max-priority-fee-per-gas"
```
[the remark this answers is not in the document]
Comment (zctocm) [reply .yarn/versions/7dea276f.yml]: done

File: .yarn/versions/7dea276f.yml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+releases:
+  "@fluent-wallet/estimate-tx": patch
+
+declined:
+  - helios-background
+  - "@fluent-wallet/consts"
+  - "@fluent-wallet/eth-transaction-schema"
+  - helios-popup
+  - "@fluent-wallet/eth_estimate-1559-fee"
+  - "@fluent-wallet/eth_fee-history"
```
[the remark this answers is not in the document]
Comment (zctocm) [reply .yarn/versions/7dea276f.yml]: done

File: .yarn/versions/7dea276f.yml [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+  - "@fluent-wallet/eth_max-priority-fee-per-gas"
+  - "@fluent-wallet/eth_sign-transaction"
```
[the remark this answers is not in the document]
Comment (zctocm) [reply .yarn/versions/7dea276f.yml]: done

File: packages/rpcs/eth_signTransaction/index.js [hunk 9]
```diff
@@ -181,6 +195,7 @@ export const main = async args => {
 
     if (dryRun)
       pk = '0x0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef'
+    console.info('txtx', toEthersTx(newTx))
```
[the remark this answers is not in the document]
Comment (zctocm) [reply packages/rpcs/eth_signTransaction/index.js]: done

File: packages/rpcs/wallet_sendTransaction/index.js [hunk 10]
```diff
@@ -85,6 +85,9 @@ export const main = async ({
       throw InvalidParams(`Invalid from address in tx ${from}`)
 
     delete params[0].nonce
+    if (params[0].type === '0x2' && params[0].gasPrice) {
```
[the remark this answers is not in the document]
Comment (zctocm) [reply packages/rpcs/wallet_sendTransaction/index.js]: done

File: .yarn/versions/7dea276f.yml (hunk 5 above)
Comment (zctocm) [reply .yarn/versions/7dea276f.yml]: done

**Review by yqrashawn — APPROVED** [review]

**Pull request merged**